src/app/screen-snippet-handler.ts

```diff
@@ -87,19 +88,37 @@ class ScreenSnippet {
    *
    * @param webContents {WeContents}
    */
-  public async capture(
-    webContents: WebContents,
-    currentWindow?: string,
-    hideOnCapture?: boolean,
-  ) {
+  public async capture(webContents: WebContents, hideOnCapture?: boolean) {
+    const currentWindowObj = BrowserWindow.getFocusedWindow();
+    const windowObj = winStore.getWindowStore();
+    const currentWindow = (currentWindowObj as ICustomBrowserWindow)?.winName;
     const mainWindow = windowHandler.getMainWindow();
+
+    if (windowObj.windowIds.length < 1) {
+      const allWindows = BrowserWindow.getAllWindows();
+      const arr: string[] = ['main'];
+
+      allWindows.forEach((window) => {
+        if (
+          (window as ICustomBrowserWindow).winName !== currentWindow ||
+          (window as ICustomBrowserWindow).winName !== 'main'
+        ) {
+          arr.push((window as ICustomBrowserWindow).winName);
+        }
+      });
+
+      arr.push(currentWindow);
+      winStore.setWindowStore({
+        windowIds: arr,
+      });
+    }
+
     if (hideOnCapture) {
-      const curWindow = getWindowByName(currentWindow || '');
-      const mainWindow = windowHandler.getMainWindow();
-      mainWindow?.minimize();
-      if (currentWindow !== 'main') {
-        curWindow?.minimize();
-      }
+      const currentWindows = BrowserWindow.getAllWindows();
+
+      currentWindows.forEach((currentWindow) => {
+        currentWindow?.minimize();
```

Comment:
minmize and focus will switch back to previously working windows.
focus will bring it back instead of doing anything else. (No config modified)

Reply:
@sbenmoussati 
You are right. The status of currently minimized (The fullscreen thing i guess it is unaffected, since I will only focus back on the windows without changing anything). Im going to add an additional case to not restore the minimized ones